default/python/universe_generation/empires.py

```diff
@@ -98,10 +128,11 @@ def compile_home_system_list(num_home_systems, systems):
     min_jumps = min(10, max(int(float(len(systems)) / float(num_home_systems * 2)), 5))
     # try to find the home systems, decrease the min jumps until enough systems can be found, or the min jump distance
     # gets reduced to 0 (meaning we don't have enough systems to choose from at all)
+    additional_reqs = homesystem_vicinity_filter if _FILTER_HOMESYSTEMS_BY_VICINITY else None
```

Comment:
You use constant `_FILTER_HOMESYSTEMS_BY_VICINITY` as condition (in two places).  It will aways evaluate to same result.